### ISSUE TITLE
unexpected exit 

### ISSUE THREAD

**Issue by christopher-schroeder** [bwa-mem2/bwa-mem2#87]
Hello,
I have a rather difficult problem. I am aligning several wgs samples, each about 100gb filesize, and for one of them bwa-mem2 avx just stops at 17gb without any error. bwa-mem2 sse4.1 stops at the same position with the following error.

`[0000][PE] skip orientation RR
[0000] 3. Calling kt_for - worker_sam
        [0000][ M::mem_process_seqs] Processed 5868668 reads in 2703.295 CPU sec, 32.876 real sec
[0000] read_chunk: 880000000, work_chunk_size: 880000219, nseq: 5868866
        [0000][ M::kt_pipeline] read 5868866 sequences (880000219 bp)...
[0000] Calling mem_process_seqs.., task: 34
[0000] 1. Calling kt_for - worker_bwt
core.exception.RangeError@BioD/bio/std/hts/bam/read.d(226): Range violation
`
Unfortunately I cannot provide a minimum example. If I reduce the fastq files to just a million around this "critical point" it runs perfectly fine and produces the expected values.
I am willing to debug something, if you would like me to.

Best
Christo

ps: bwa-mem (the normal old version) is able to process the sample without any errors.

**Comment by keiranmraine** [comment]
The error message is raised by the BioD package, bwa itself doesn't read or write bam files.  Although you'd expect the surrounding tools to behave the same are you sure the issue is with bwa-mem2 itself and not something feeding in to or reading from bwa?

**Comment by christopher-schroeder** [comment]
I will rerun and pipe to /dev/null to see what happens

**Comment by keiranmraine** [comment]
If you have sufficient space I'd recommend the following while determining if BWA is the source of the issue

```
bwa-mem .... | gzip -c -2 > sam.gz
```

Then you can also see if the issue remains in the rest of the flow via:

```
zcat sam.gz | <other tools>
```

This may also help identify if an individual read is malformed by slicing up that file later.  Possibly also worth determining if the remainder of the flow works with the samtools implementations of the downstream processing (if possible).

**Comment by christopher-schroeder** [comment]
I used pigz to speed up the compression process. The command was

`
(bwa-mem2 mem -t 88 -R '@RG\tID:1875-0_Blood\tSM:1875-0_Blood\tPL:ILLUMINA' resources/genome.fasta results/merged/1875-0_Blood_R1.fastq.gz results/merged/1875-0_Blood_R2.fastq.gz | pigz -c > 1875-0_Blood.sam.gz) 2> 1875-0_Blood.log
`
I counted the lines

`zcat 1875-0_Blood.sam.gz | wc -l`

and got 195,759,052 lines. 

A count of the bwa aligned file

`sambamba view -c 1875-0_Blood.bwa.bam`

shows 2,239,533,143 aligned reads (including split map secondary reads)

**Comment by keiranmraine** [comment]
Can you run with `-v 4`:

```
   -v INT        verbose level: 1=error, 2=warning, 3=message, 4+=debugging [3]
```

The last read of the sam.gz file is missing the AUX tags, a possible place to investigate for the devs.

**Comment by keiranmraine** [comment]
(/dev/null for bwa stdout)

**Comment by christopher-schroeder** [comment]
Running, but it will take a lot of time. The debug mode is very slow.

**Comment by mp15** [comment]
Another way you can check for undefined behaviour is to compile with `-fsanitize=address`. This is a bit faster but may catch if the issue is caused by a bad memory read, you can breakpoint on __asan_on_error@plt to catch the error before it writes over your stack. The final alternative is valgrind which should work on sse41 but is rather slow.

**Comment by christopher-schroeder** [comment]
I dont even know what that means OO

**Comment by christopher-schroeder** [comment]
here is tail -n from the debug log
[1875-0_Blood.v4.tail.log](https://github.com/bwa-mem2/bwa-mem2/files/5073710/1875-0_Blood.v4.tail.log)

**Comment by yuk12** [comment]
@christopher-schroeder are you still facing the problem? 
What is the memory capacity of this system?
can you try and check with smaller number of threads?

**Comment by christopher-schroeder** [comment]
yes, the error remain. I had no choice and switched back to bwa mem for the project and everything was mapped and downstream analysis are running currently.
The memory capacity of the system is 756gb, that should be no issue.
Sure, what number do you have in mind @yuk12? 44?

**Comment by yuk12** [comment]
The available memory is sufficient to run mem2. You can start with 16 threads. The log shows that it is hanged in the seeding stage.
Also, you can try the latest commit, where index size is reduced significantly, and see.

**Comment by christopher-schroeder** [comment]
I tried 16 cores with the newest version and it worked. I increase it to 32 now.

**Comment by christopher-schroeder** [comment]
with 32 threads a get the following error:
```
bwa-mem2: src/bwamem.cpp:904: int mem_kernel1_core(FMI_search*, const mem_opt_t*, bseq1_t*, int, mem_chain_v*, mem_seed_t*, int64_t, mem_cache*, int): Assertion `num_smem < *wsize_mem' failed.
```

**Comment by skchronicles** [comment]
@christopher-schroeder Do you have any updates for this issue? I am also running into the same problem.

**Comment by christopher-schroeder** [comment]
I dont know the technical background but I now used -K 500000000 as parameter for a few month on 100+ whole genomes and never got this error again. So I guess it this is at least a workaround.

**Comment by skchronicles** [comment]
@christopher-schroeder Thank you for the quick reply! I appreciate. I will let you know if this also fixes our issue too.

**Comment by christopher-schroeder** [comment]
@skchronicles Did it work?
